src/schema-association-service-1espt.ts

```diff
@@ -0,0 +1,92 @@
+            
+            // Using getItem from GitApi: getItem(repositoryId: string, path: string, project?: string, scopePath?: string, recursionLevel?: GitInterfaces.VersionControlRecursionType, includeContentMetadata?: boolean, latestProcessedChange?: boolean, download?: boolean, versionDescriptor?: GitInterfaces.GitVersionDescriptor, includeContent?: boolean, resolveLfs?: boolean, sanitize?: boolean): Promise<GitInterfaces.GitItem>;
+            var schemaFile = await gitApi.getItem(repository.id, "schema/1espt-base-schema.json", "1ESPipelineTemplates", undefined, undefined, true, true, true, undefined, true, true)
+
+            const schemaContent = schemaFile.content;
+            const schemaUri = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+            await vscode.workspace.fs.writeFile(schemaUri, Buffer.from(schemaContent));
+            return schemaUri;
+        }
+        else
+        // if user is signed in with account other than microsoft, then disable 1ESPT schema and delete the 1ESPT schema file
+        {
+            const config = vscode.workspace.getConfiguration('azure-pipelines', workspaceFolder)
+            config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.Workspace);
+            await vscode.workspace.fs.delete(Utils.joinPath(context.globalStorageUri, '1ESPTSchema'), { recursive: true })
+        }
+    }
+    catch (error) {
+        logger.log(`Error : ${error} while fetching 1ESPT schema for org: ${organizationName}  : `, 'SchemaDetection');
+    }
+    return undefined;
+}
+
+export async function getCached1ESPTSchemaInformation(context: vscode.ExtensionContext, organizationName: string, session: AzureSession, oneesptSchemaEnabled: boolean, lastUpdated1ESPTSchema: Map<string,Date>, seen1ESPTOrganizations: Set<string>): Promise<[URI, boolean]> {
+    var skipOrgSpecificCachedSchema = false
+    var validSessionToEnable1ESPTSchema = false;
+    if (seen1ESPTOrganizations.has(organizationName)) {
+        const schemaUri1ESPT = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+        validSessionToEnable1ESPTSchema = session.userId.endsWith("@microsoft.com") || session.userId.endsWith(".microsoft.com")
+
+        try
+        {
+            // check if 1ESPT schema file exists
+            var schemaFileExists = await vscode.workspace.fs.readFile(schemaUri1ESPT);
+            if (schemaFileExists) {
+            // Fetch cached 1ESPT schema if:
+            // 1) User is signed in with microsoft account
+            // 2) 1ESPT schema is enabled
+            // 3) last fetched 1ESPT schema is less than 24 hours old
+                if (validSessionToEnable1ESPTSchema && oneesptSchemaEnabled && ((new Date().getTime() - lastUpdated1ESPTSchema.get(organizationName).getTime()) < milliseconds24hours)) {
```

Comment:
Nits: For performance, it's better to perform these checks (which are really cheap) first, before checking for the existence of the schema file on-disk.